src/main/java/io/vertx/core/Future.java

```diff
@@ -479,6 +481,71 @@ default Context context() {
     return (Future<T>) AsyncResult.super.otherwiseEmpty();
   }
 
+  /**
+   * Bridges this Vert.x future to a {@link CompletionStage} instance.
+   * <p>
+   * The {@link CompletionStage} handling methods will be called from the thread that resolves this future.
+   *
+   * @return a {@link CompletionStage} that completes when this future resolves
+   */
+  @GenIgnore(GenIgnore.PERMITTED_TYPE)
+  default CompletionStage<T> toCompletionStage() {
+    CompletableFuture<T> completableFuture = new CompletableFuture<>();
+    this.setHandler(ar -> {
+      if (ar.succeeded()) {
+        completableFuture.complete(ar.result());
+      } else {
+        completableFuture.completeExceptionally(ar.cause());
+      }
+    });
+    return completableFuture;
+  }
+
+  /**
+   * Bridges a {@link CompletionStage} object to a Vert.x future instance.
+   * <p>
+   * The Vert.x future handling methods will be called from the thread that completes {@code completionStage}.
+   *
+   * @param completionStage a completion stage
+   * @param <T>             the result type
+   * @return a Vert.x future that resolves when {@code completionStage} resolves
+   */
+  @GenIgnore(GenIgnore.PERMITTED_TYPE)
+  static <T> Future<T> from(CompletionStage<T> completionStage) {
```

Comment:
can you rename to `fromCompletionStage` ?

Reply:
@vietj I had the same comment at first but then I agreed with @jponge that a compact form is more readable.

Reply:
in a previous conversation with @jponge we agreed on `fromCompletionStage` (see https://github.com/vert-x3/wiki/wiki/RFC:-Future-API#open-questions--todo) when we drafted this API. I think the main reason is that it was symetrical to `toCompletionStage` and be more readable for `CompletionStage` users

Reply:
When we pair-programmed with @tsegismont I thought `from` was nice. But I can rename.